src/docfx/build/page/PageBuilder.cs

```diff
@@ -261,6 +266,27 @@ private SystemMetadata CreateSystemMetadata(ErrorBuilder errors, FilePath file,
         return systemMetadata;
     }
 
+    private string? GetCanonicalUrlWithMonikerIfNecessary(string? pageType, FilePath file)
+    {
+        if (pageType == "profile")
```

Comment:
is this special logic come from docs.ui as well?

Reply:
Can you double confirm that for `Profile`, no canonical URL is needed to generate?

Reply:
sure

Reply:
@herohua  
I just check the output result of https://github.com/MicrosoftDocs/DocsRoot/blob/972fb584bff2eb20b14a649fe41ed4950d74c04f/profile/edit.md?plain=1#L3 
there is no canonical_url generated, which means the scenario exists. 

But I am not sure whether the business logic is still valid, but I think that is out of scope of this task. 

I guess we would better to keep the logic to avoid breaking something and we may investigate it later as we have some other urgent tasks in progress.